crates/consensus/src/transaction/eip4844/builder.rs

```diff
@@ -0,0 +1,406 @@
+#[cfg(not(feature = "kzg"))]
+use alloy_eips::eip4844::Blob;
+#[cfg(feature = "kzg")]
+use c_kzg::{Blob, KzgCommitment, KzgProof, KzgSettings};
+
+use alloy_eips::eip4844::{BYTES_PER_BLOB, FIELD_ELEMENTS_PER_BLOB};
+
+use super::utils::WholeFe;
+
+/// A builder for creating a [`BlobTransactionSidecar`].
+///
+/// [`BlobTransactionSidecar`]: crate::BlobTransactionSidecar
+#[derive(Debug, Clone)]
+pub struct PartialSidecar {
+    /// The blobs in the sidecar.
+    blobs: Vec<Blob>,
+    /// The number of field elements that we have ingested, total.
+    fe: usize,
+}
+
+impl Default for PartialSidecar {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl PartialSidecar {
+    /// Create a new builder, and push an empty blob to it. This is the default
+    /// constructor, and allocates space for 2 blobs (256 KiB). If you need more
+    /// space, use [`PartialSidecar::with_capacity`].
```

Comment:
> ```
> If you need more space, use [`PartialSidecar::with_capacity`].
> ```
I would word this as `if you want to pre-allocate capacity`, since `ingest_valid_fe` and other pushing ops would increase the capacity anyways